### PR TITLE
YAML compatibility

### PR DESCRIPTION
**Pull request by Poofjunior**
We can now use configs written as either YAML or TOML files.

I found a yaml reader/writer that:
* preserves order
* preserves comments

Given that we have this working, I suggest we switch over to using YAMLs for our configs. To make that easy, this code can actually load TOMLs and save them as YAMLs if you rune `config.save("config.yaml")` where the file saved has a `".yaml"` file extension--even if it was loaded from a TOML file. Otherwise, you can use this script (pasted at the end) to convert our existing TOML configs to a YAML config.

No changes are needed in the other repos to merge this request as-is.

### Testing
I added a test file that you run with pytest. It tests loading/saving/member-access.

I can also confirm that the exa-spim-control code will simulate when running a yaml-style config.

### TOML to YAML conversion script:
````python
#!/usr/bin/env python3

import toml
from pathlib import Path
from ruamel.yaml import YAML
import argparse


def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("toml_path", type=str, default=None,
                        help="input path to a toml file.")
    parser.add_argument("--yaml_path", type=str, default=None,
                        help="output path of the yaml file. " +
                             "Defaults to same folder as the toml file.")
    args = parser.parse_args()

    toml_path = Path(args.toml_path)
    if not toml_path.is_file():
        raise RuntimeError("Error: toml path must point to a toml file.")
    if args.yaml_path is None:
        yaml_path = Path(toml_path.parent /
                         Path(str(toml_path.name).rstrip("toml") + "yaml"))
    else:
        yaml_path = Path(args.yaml_path)
        if not yaml_path.is_file():
            raise RuntimeError("Error: yaml path must end with the yaml file "
                               f"to be created.\r\nPath is: {str(yaml_path)}")

    # Do the conversion.
    with open(str(toml_path), 'r') as toml_file:
        cfg_file = toml.load(toml_file)
        with open(str(yaml_path), 'w') as yaml_file:
            yaml = YAML()
            yaml.dump(cfg_file, yaml_file)


if __name__ == "__main__":
    main()
````

### CODE REVIEW

File: tests/test_configs/config.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  subject_id: sim_test
+  tile_prefix: tile
+  tile_overlap_x_percent: 15
+  tile_overlap_y_percent: 10
+  z_step_size_um: 1
+  volume_x_um: 10615.616
+  volume_y_um: 7958.7
+  volume_z_um: 100
+  laser_wavelengths:
+  - 561  # test comment!
```
Comment (adamkglaser) [inline tests/test_configs/config.yaml:13]: Just for my own learning. Why does 561 need a hyphen here but not other inputs?

File: tests/test_configs/config.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    '638': 4
+    '561': 13
+    '405': 6
+    galvo_a: 17
+    galvo_b: 19
+tiger_controller_driver_kwds:
+  com_port: COM3
+estimates: {}
+channel_specs:
+  '488':
```
Comment (adamkglaser) [inline tests/test_configs/config.yaml:64]: Similarly here, quotes are needed to explicitly say that 488 is a string and not a number?

**Review by adamkglaser — COMMENTED** [review]
Looks great that was fast! Just a few questions on YAML formatting.

File: tests/test_configs/config.yaml (hunk 1 above)
Comment (Poofjunior) [reply tests/test_configs/config.yaml:13]: Oh; this is one of two valid formats for documenting lists. What you see here is also equivalent to:
````yaml

laser_wavelengths: [561]
````
Note from [Wikipedia](https://en.wikipedia.org/wiki/YAML#Basic_components)

File: tests/test_configs/config.yaml (hunk 2 above)
Comment (Poofjunior) [reply tests/test_configs/config.yaml:64]: Totally. 
YAML format "autodetects" simple types. That is to say, a list of integers would be interpretted as a list of integers unless you explicitly say that they're strings. In our old TOML config, the laser channel specs keys were actually in string format, so this is preserved here.

**Comment by adamkglaser** [comment]
Got it!

**Comment by adamkglaser** [comment]
Fine by me to merge

**Pull request merged**